models.py

```diff
@@ -432,47 +463,111 @@ def sample(
                 running mini-MPNN. None, 'linear', or 'cosine'
             use_fullmpnn: use "full" ProteinMPNN at each step.
             use_fullmpnn_for_final: use "full" ProteinMPNN at the final step.
-            use_reconstruction_guidance: use reconstruction guidance on the conditioning.
-            use_classifier_free_guidance: use classifier-free guidance on the conditioning.
-            guidance_scale: weight for reconstruction/classifier-free guidance.
-            noise_schedule: specify the noise level timesteps for sampling.
+            num_seqs: number of sequences to sample at the last step, using ProteinMPNN.
+            mpnn_batch_size: batch size for ProteinMPNN.
+
+            # Auxilliary arguments
             tqdm_pbar: progress bar in interactive contexts.
             return_last: return only the sampled structure and sequence.
             return_aux: return a dict of everything associated with the sampling run.
         """
 
-        def ode_step(sigma_in, sigma_next, xt_in, x0_pred, gamma, guidance_in=None):
-            if gamma > 0:
-                t_hat = sigma_in + gamma * sigma_in
-                sigma_delta = torch.sqrt(t_hat**2 - sigma_in**2)
-                noisier_x = xt_in + utils.expand(
-                    sigma_delta, xt_in
-                ) * noise_scale * torch.randn_like(xt_in).to(xt_in)
-                xt_in = noisier_x * utils.expand(seq_mask, noisier_x)
-                sigma_in = t_hat
+        def get_denoiser(forward_fn, seq_mask, residue_index):
+            def x0_from_xt(
+                xt,
+                sigma,
+                x_self_cond,
+                s_self_cond,
+                crop_cond_coords,
+                run_mpnn,
+                use_mpnn,
+            ):  #! added use_mpnn
+                x0, s_logprobs, x_self_cond, s_self_cond = forward_fn(
+                    noisy_coords=xt,
+                    noise_level=sigma,
+                    seq_mask=seq_mask,
+                    residue_index=residue_index,
+                    struct_self_cond=x_self_cond,
+                    struct_crop_cond=crop_cond_coords,
+                    seq_self_cond=s_self_cond,
+                    run_mpnn_model=run_mpnn,
+                    use_mpnn_model=use_mpnn,
+                )
+                return x0, s_logprobs, x_self_cond, s_self_cond
+
+            return x0_from_xt
+
+        def ode_step(
+            sigma_in,
+            sigma_next,
+            xt_in,
+            x0_pred,
+            guidance_in=None,
+            return_addl=False,
+            sidechain_mode=False,
+            gamma=0.0,
+        ):
+            if sidechain_mode == False:
+                pass
+            elif sidechain_mode == True:
+                if gamma > 0:
```

Comment:
i think this can all be replaced with just one statement: `if sidechain_mode and gamma > 0:`